### DISCUSSION TITLE
PyLance suddenly not detecting libraries installed via pip.

### DISCUSSION THREAD

**Discussion by ItsCubeTime** [microsoft/pylance-release#4178] [Q&A]
Hi, Im using PyLance with Visual Studio Code. Autocompletion & syntax highlighting is working as expected for Python STD modules, but for anything installed via `pip install X` I wont get any autocompletion for. Used to work fine last week (which was last I coded in Python with VS Code).
```py
import math
math.
```
would give me suggestions for math & lets me enter the math modules __init__ file by ctrl clicking it
```py
import numpy
numpy.
```
gives me no suggestions & ctrl clicking brings me nowhere.

Setting the language server to Jedi in the Python extensions settings fixes the issue, so it only affects Pylance it seems.

In my settings.json I have:
```
"python.defaultInterpreterPath": "C:\\Users\\olliv\\AppData\\Local\\Programs\\Python\\Python310\\python.exe",
```
The packages were installed via:
```
pip install numpy
... (success)
pip --version
pip 23.0.1 from C:\Users\olliv\AppData\Local\Programs\Python\Python310\lib\site-packages\pip (python 3.10)
```
Running a py file making use of pip installed libraries using `C:/Users/olliv/AppData/Local/Programs/Python/Python310/python.exe <filename>` works fine.

**Comment by rchiodo** [comment]
Are you opening a folder first or is this just in a loose file? Might be this problem:
https://github.com/microsoft/pylance-release/issues/4162

**Reply by ItsCubeTime** [reply]
Not sure if its related but, it seems the issue only occurs when I dont have a workspace or folder open in VS Code. In other words, opening up a folder resolved the issue.

This is the Python Language Server logs reported by VS Code:
```
[Info  - 10:31:29 PM] (35804) Pylance language server 2023.3.40 (pyright 04471971) starting
[Info  - 10:31:29 PM] (35804) Server root directory: c:\Users\olliv\.vscode\extensions\ms-python.vscode-pylance-2023.3.40\dist
[Info  - 10:31:29 PM] (35804) Auto-indent enabled
[Info  - 10:31:29 PM] (35804) virtual workspace: 
[Info  - 10:31:29 PM] (35804) Starting service instance "<default>"
[Info  - 10:31:29 PM] (35804) No pyproject.toml file found.
[Info  - 10:31:29 PM] (35804) Setting pythonPath for service "<default>": "C:\Users\olliv\AppData\Local\Programs\Python\Python310\python.exe"
[Warn  - 10:31:29 PM] (35804) stubPath typings is not a valid directory.
[Info  - 10:31:29 PM] (35804) Assuming Python version 3.10
[Info  - 10:31:29 PM] (35804) Assuming Python platform Windows
[Info  - 10:31:29 PM] (35804) Searching for source files
[Info  - 10:31:29 PM] (35804) No source files found.
[Info  - 10:31:29 PM] (35804) No pyproject.toml file found.
[Info  - 10:31:29 PM] (35804) Setting pythonPath for service "<default>": "C:\Users\olliv\AppData\Local\Programs\Python\Python310\python.exe"
[Warn  - 10:31:29 PM] (35804) stubPath typings is not a valid directory.
[Info  - 10:31:29 PM] (35804) Assuming Python version 3.10
[Info  - 10:31:29 PM] (35804) Assuming Python platform Windows
[Info  - 10:31:29 PM] (35804) Searching for source files
[Info  - 10:31:29 PM] (35804) No source files found.
```
![image](https://user-images.githubusercontent.com/20190653/229224534-ec324325-38de-48dd-a7ef-8eb6ce5e2004.png)

**Comment by rchiodo** [comment]
Your logs for pylance would be very helpful in determining where Pylance is looking for numpy.

My guess is the 'defaultInterpreterPath' setting. I believe that's no longer used by the Python extension. You might have a different python 'picked' in the bottom right:

![image](https://user-images.githubusercontent.com/19672699/229176000-b2197848-213e-4168-9019-4187a2e0b146.png)

**Comment by rchiodo** [comment]
If it only happens with no folder open, then this should be #4142
